sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -582,9 +583,29 @@ class SparkSqlAstBuilder extends AstBuilder {
   }
 
   override def visitSetVariable(ctx: SetVariableContext): LogicalPlan = withOrigin(ctx) {
-    val variableName = visitMultipartIdentifier(ctx.multipartIdentifier)
-    val expression = ctx.expression().toString
-    SetVariableCommand(VariableIdentifier(variableName.last), expression)
+    if (ctx.assignmentList().isEmpty) {
+      SetVariableCommand(Seq(VariableIdentifier("var")), "7")
+    } else {
+      val assignCtx = ctx.assignmentList()
+      val varList = assignCtx.assignment().asScala.map {
+        assign => {
+          val varname = visitMultipartIdentifier(assign.key)
+          VariableIdentifier(varname.head)
+        }
+      }
+
+      val assignments = assignCtx.assignment()
+      val exprStrList = assignments.asScala.map {
+        assign => {
```

Comment:
nit: no need for `{`. I'd also move assign to the line above (as more Scala-idiomatic IMHO)

Reply:
addressed